### DISCUSSION TITLE
Filter or sorting calculated Columns

### DISCUSSION THREAD

**Discussion by akamasAT** [Power-Components/livewire-powergrid#541] [Q&A]
I have a manually created column with no corresponding database field (its a date, that gets calculated), is there a way to make this column sortable or filterable?

**Comment by akulmehta** [comment]
Could you share some code.

**Reply by akamasAT** [reply]
just adding a column. the function calculateContractEndDate() returns a Carbon date

`->addColumn('contract_until', fn (ProductData $model) => $model->calculateContractEndDate()->format('d.m.Y'))`

since the table sorts serverside, trying to sort by this column ends in an exception:
`SQLSTATE[42S22]: Column not found: 1054 Unknown column 'contract_until' in 'order clause'`

**Comment by akulmehta** [comment]
Sorting would have to be done serverside as powergrid paginates the results. Filtering would also need to be done by the database as it involves SQL queries, much more performant. Can the `calculatedcontractenddate` not be saved in the DB?

**Reply by akamasAT** [reply]
yes, but actually no. I guess using a collection instead of a model would do the trick, but thank you!

**Reply by ashek1412** [reply]
hi @akamasAT , have u solved it using collection?
